File: Patterns/Facade Pattern/Cpp/HomeTheater/DvdPlayer.cpp
#pragma once

#include <iostream>
#include <string>

using namespace std;

class DvdPlayer {
private:
    string description;
    int currentTrack;
    string movie;

public:
    DvdPlayer(string description) : description{description} {}
    
    void on() {
        cout << description << " on\n";
    }

    void off() {
        cout << description << " off\n";
    }

    void eject() {
        movie = "";
        cout << description << " eject\n";
    }

    void play(string movie) {
        this->movie = movie;
        currentTrack = 0;
        cout << description << " playing \"" << movie << "\"\n";
    }

    void play(int track) {
        if (movie.empty()) {
            cout << description << " can't play track " << currentTrack << ", no dvd inserted\n";
        }
        else {
            currentTrack = track;
            cout << description << " playing track " << currentTrack << " of \"" << movie << "\"\n";
        }
    }

    void stop() {
        currentTrack = 0;
        cout << description << " stopped \"" << movie << "\"\n";
    }

    void pause() {
        cout << description << " paused \"" << movie << "\"\n";
    }

    void setTwoChannelAudio() {
        cout << description << " set two channel audio\n";
    }

    void setSurroundAudio() {
        cout << description << "set surround audio\n";
    }

    string toString() {
        return description;
    }
};

File: Patterns/Facade Pattern/CSharp/HomeTheater/DvdPlayer.cs
using System;

namespace HomeTheater
{
    public class DvdPlayer
    {
        private string description;
        private int currentTrack;
        private string? movie;

        public DvdPlayer(string description)
        {
            this.description = description;
        }

        public void On() => Console.WriteLine($"{description} on");

        public void Off() => Console.WriteLine($"{description} off");

        public void Eject()
        {
            movie = null;
            Console.WriteLine($"{description} eject");
        }

        public void Play(string movie)
        {
            this.movie = movie;
            currentTrack = 0;
            Console.WriteLine($"{description} playing \"{movie}\"");
        }

        public void Play(int track)
        {
            if (movie == null)
            {
                Console.WriteLine($"{description} can't play track {currentTrack}, no cd inserted");
            }
            else
            {
                currentTrack = track;
                Console.WriteLine($"{description} playing track {currentTrack}");
            }
        }

        public void Stop() {
            currentTrack = 0;
            Console.WriteLine($"{description} stopped");
        }

        public void Pause() => Console.WriteLine($"{description} pause \"{movie}\"");

        public void SetTwoChannelAudio() => Console.WriteLine($"{description} set two channel audio");

        public void SetSurroundAudio() => Console.WriteLine($"{description} set surround audio");

        public string Description { get => description; private set => description = value; }
    }
}